lib/mrkt/concerns/crud_leads.rb

```diff
@@ -13,22 +15,30 @@ def get_leads(filter_type, filter_values, fields: nil, batch_size: nil, next_pag
     end
 
     def createupdate_leads(leads, action: 'createOrUpdate', lookup_field: nil, partition_name: nil, async_processing: nil)
-      post('/rest/v1/leads.json') do |req|
-        params = {
-          action: action,
-          input: leads
-        }
-        params[:lookupField] = lookup_field if lookup_field
-        params[:partitionName] = partition_name if partition_name
-        params[:asyncProcessing] = async_processing if async_processing
+      # see also: http://developers.marketo.com/documentation/rest/createupdate-leads/
+      # > You can update a maximum of 300 leads with a single API call.
+      leads.each_slice(MAX_LEADS_SIZE).map do |_leads|
```

Comment:
> the output to not break the API

Agree.
But when leads is over 300, we call multiple requests to marketo and receive multiple responses.